src/main/java/uk/gov/laa/ccms/caab/service/AssessmentService.java

```diff
@@ -0,0 +1,559 @@
+              .blockOptional()
+              .orElse(statusKey);
+
+      if (MEANS.getName().equalsIgnoreCase(assessment.getName())) {
+        application.setMeansAssessmentStatus(statusValue);
+      } else if (MERITS.getName().equalsIgnoreCase(assessment.getName())) {
+        application.setMeritsAssessmentStatus(statusValue);
+      }
+    }
+  }
+
+  /**
+   * Determines if the assessment of an application has been amended based on the type of assessment
+   * provided, whether means or merits.
+   *
+   * @param application the application to check for amendments
+   * @param assessment the assessment details to determine the type
+   * @return true if the specified assessment type has been amended; false otherwise
+   */
+  private boolean isApplicationsAssessmentAmended(
+      final ApplicationDetail application,
+      final AssessmentDetail assessment) {
+
+    if (assessment != null) {
+      if (MEANS.getName().equalsIgnoreCase(assessment.getName())) {
+        return application.getMeansAssessmentAmended();
+      } else if (MERITS.getName().equalsIgnoreCase(assessment.getName())) {
+        return application.getMeritsAssessmentAmended();
+      }
+    }
+    return false;
+  }
+
+
+  /**
+   * Determines if a reassessment is required based on changes between an
+   * application and its latest assessment.
+   *
+   * @param application the application details to compare against the assessment
+   * @param assessment the assessment details to compare against the application
+   * @return true if a reassessment is necessary; false otherwise
+   */
+  protected boolean isReassessmentRequired(
+      final ApplicationDetail application,
+      final AssessmentDetail assessment) {
+
+    if (!application.getAmendment()) {
+      if (assessment != null) {
+
+        final AssessmentEntityTypeDetail assessmentEntityType =
+            getAssessmentEntityType(assessment, PROCEEDING.getType());
+
+        //check for any proceeding attribute changes between assessment and application
+        if (checkAssessmentForProceedingKeyChange(application, assessmentEntityType)) {
+          return true;
+        }
+
+        //check if proceeding is deleted
+        if (application.getProceedings() != null) {
+          log.debug("app.getProceedings().size(): {} - opaListEntity.getOpaEntities().size(): {}",
+              application.getProceedings().size(), assessmentEntityType.getEntities().size());
+
+          if (application.getProceedings().size() < assessmentEntityType.getEntities().size()) {
+            log.debug("When proceeding is deleted condition - returns TRUE");
+            return true;
+          }
+        }
+
+        //used to check if the proceeding is deleted - this should be included for both assessments,
+        //although not specifically needed for means, both assessments and application data should
+        //be kept in sync.
+        if (Boolean.TRUE.equals(application.getMeritsReassessmentRequired())) {
```

Comment:
Don't really follow how the comment relates to the isMeritsReassessmentRequired check.

Reply:
This value from what i can find is only set when the proceeding is deleted